### ISSUE TITLE
Cannot compile on Artix/Arch Linux

### ISSUE THREAD

**Issue by invader-zimm** [cropinghigh/stdcdec#2]
I'm using Artix Linux, all packages up to date, and compilation fails with this error:
```
[ 16%] Building CXX object CMakeFiles/stdc_demod.dir/stdc_demod.cpp.o
[ 33%] Linking CXX executable stdc_demod
[ 33%] Built target stdc_demod
[ 50%] Building CXX object CMakeFiles/stdc_decoder.dir/stdc_decoder.cpp.o
/home/user/Downloads/github/inmarsat/stdcdec/stdc_decoder.cpp: In function ‘void sendDecodedFrameViaUdp(inmarsatc::decoder::Decoder::decoder_result, int, sockaddr_in)’:
/home/user/Downloads/github/inmarsat/stdcdec/stdc_decoder.cpp:122:31: error: invalid use of incomplete type ‘struct std::array<char, 664>’
  122 |     auto serialized = to_bytes(data);
      |                       ~~~~~~~~^~~~~~
In file included from /usr/include/c++/12.1.0/mutex:38,
                 from /usr/include/inmarsatc_demodulator.h:22,
                 from /usr/include/inmarsatc_decoder.h:20,
                 from /home/user/Downloads/github/inmarsat/stdcdec/stdc_decoder.cpp:2:
/usr/include/c++/12.1.0/tuple:1595:45: note: declaration of ‘struct std::array<char, 664>’
 1595 |   template<typename _Tp, size_t _Nm> struct array;
      |                                             ^~~~~
/home/user/Downloads/github/inmarsat/stdcdec/stdc_decoder.cpp: In instantiation of ‘std::array<char, sizeof (T)> to_bytes(const T&) [with T = inmarsatc::decoder::Decoder::decoder_result]’:
/home/user/Downloads/github/inmarsat/stdcdec/stdc_decoder.cpp:122:31:   required from here
/home/user/Downloads/github/inmarsat/stdcdec/stdc_decoder.cpp:98:32: error: return type ‘struct std::array<char, 664>’ is incomplete
   98 | std::array< char, sizeof(T) >  to_bytes( const T& object ) {
      |                                ^~~~~~~~
/home/user/Downloads/github/inmarsat/stdcdec/stdc_decoder.cpp:99:35: error: ‘std::array<char, 664> bytes’ has incomplete type
   99 |     std::array< char, sizeof(T) > bytes ;
      |                                   ^~~~~
make[2]: *** [CMakeFiles/stdc_decoder.dir/build.make:76: CMakeFiles/stdc_decoder.dir/stdc_decoder.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:113: CMakeFiles/stdc_decoder.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

```
The gcc version is 12.1.0

**Comment by cropinghigh** [comment]
Fixed by f6e6645a551b59d075d50e56853faf6bb33a3637

**Comment by invader-zimm** [comment]
Yup, all good now. Thank you.
